pkg/controller/glusternode/reconcile.go

```diff
@@ -0,0 +1,98 @@
+package glusternode
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+
+	operatorv1alpha1 "github.com/gluster/anthill/pkg/apis/operator/v1alpha1"
+
+	"github.com/gluster/anthill/pkg/reconciler"
+)
+
+var (
+	log                                         = logf.Log.WithName("controller_glusternode")
+	allProcedures      reconciler.ProcedureList = []reconciler.Procedure{*ProcedureV1}
+	reconcileProcedure *reconciler.Procedure
+	err                error
+	procedureStatus    *reconciler.ProcedureStatus
+)
+
+// Reconcile reads that state of the node for a GlusterNode object and makes changes based on the state read
+// and what is in the GlusterNode.Spec
+
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileGlusterNode) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling GlusterNode")
+
+	// Fetch the GlusterNode instance
+	instance := &operatorv1alpha1.GlusterNode{}
+	err = r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	// Get current reconcile version from CR
+	version := instance.Status.ReconcileVersion
```

Comment:
I don't think you can reconcile to a GlusterNode independently from the GlusterCluster version. I think you'll need to figure out the target version from the GlusterCluster, not from the status field on the GlusterNode.

Reply:
Moving the field to the GusterNode Spec so it can be set by the cluster.